pkg/kds/mux/server.go

```diff
@@ -131,8 +131,7 @@ func (s *server) Start(stop <-chan struct{}) error {
 	grpcOptions = append(
 		grpcOptions,
 		grpc.ChainUnaryInterceptor(s.unaryInterceptors...),
-		//nolint:staticcheck
-		grpc.ChainUnaryInterceptor(otelgrpc.UnaryServerInterceptor()),
+		grpc.StatsHandler(otelgrpc.NewServerHandler()),
```

Comment:
I remember us rolling back to this because of an issue in grpc. This is fixed now?

Reply:
Ah didn't realize that. In fact yes [the issue](https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4576) seems to have been fixed by https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4577

Reply:
which was released in the version we're using